c10/core/Event.h

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include "ATen/Context.h"
+#include "c10/util/Optional.h"
+#include "c10/util/Exception.h"
+#include "c10/core/Stream.h"
+
+#include <memory>
+#include <limits>
+#include <atomic>
+#include <utility>
+
+namespace at { namespace cuda {  namespace detail {
```

Comment:
Ugh, layering violation XD (I am not sure if we've merged libraries to the point where this is tolerated behavior. In particular, I suspect that this change here will break mobile in fbcode, because I'm pretty sure you're referring to a struct that is not synced to xplat. This is not a deep problem, because you're never going to use Event on mobile, but it's something that mechanically will need to be fixed.)

Reply:
This can be fixed by exposing the void* pointer. That will also get rid of the friend declaration.